multiple-erc20-to-erc20/contracts/contracts/upgradeable_contracts/factories/HomeBridgeFactory.sol

```diff
@@ -88,8 +88,11 @@ contract HomeBridgeFactory is BasicBridgeFactory {
         token.transferOwnership(msg.sender);
         // cast proxy as IHomeBridge
         IHomeBridge homeBridge = IHomeBridge(proxy);
+        // adjusting limits for token decimals (declaring these vars here to get around "Stack too deep" error)
```

Comment:
If you declare all the vars there it gives stack too deep? Because now it kinda looks strange that part is there and part is in-line the function call...

Reply:
Yeah but I though and it makes sense.
Here you have a function call with many arguments, so you put all the args on the stack. But the last argument is a function call with more arguments. So now you also add to the stack another function call and its args. 
So what I did is first call the inner function, and after its stack cleaned I'm calling the long function.

Reply:
I meant to do this:
```
// adjusting limits for token decimals (declaring these vars here to get around "Stack too deep" error)
uint256 homeDailyLimitVal = adjustToDefaultDecimals(homeDailyLimit(), _tokenDecimals);
uint256 homeMaxPerTxVal = adjustToDefaultDecimals(homeMaxPerTx(), _tokenDecimals);
uint256 minPerTxVal = adjustToDefaultDecimals(minPerTx(), _tokenDecimals);
uint256 foreignDailyLimitVal = adjustToDefaultDecimals(foreignDailyLimit(), _tokenDecimals);
uint256 foreignMaxPerTxVal = adjustToDefaultDecimals(foreignMaxPerTx(), _tokenDecimals);
// initialize homeBridge
homeBridge.initialize(bridgeValidators, homeDailyLimitVal, homeMaxPerTxVal, minPerTxVal, gasPrice(), requiredBlockConfirmations(), token, foreignDailyLimitVal, foreignMaxPerTxVal, homeBridgeOwner());
```

Reply:
I think it looks more "organized" this way